File: quests/erudnext/Raine_Beteria.pl
# Raine Beteria
# Location: erudnext - Erudin
# npcID: 24021
# spawngroupID: 24021
# Quests:
#    Part of quest for Incandescent Wand
#    Part of quest for Spell: Tashania   by darkonig

sub EVENT_SAY {
  if ($text=~/hail/i) {
    quest::say("Welcome. I am Raine Beteria.");
  }
}

sub EVENT_ITEM {
  # Text is completely made up
  if(($itemcount{6339} == 1) && ($gold >= 50)) {
    quest::say("Here is your wand back - I assure you it is now fully enchanted.");
    quest::summonitem(6340);
    quest::faction(211, 10);
    quest::faction(145, 10);
    quest::faction(147, 10);
    quest::exp(1000);
  }

  # Tashania Quest
  # items needed:
  #     10792 Gleaming Coin of Tash
  # items received:
  #     10793 Radiant Coin of Tash
  # plus faction received:
  #     442   Temple of Sol Ro
  # minus faction received:
  #     291   Shadowed Men
  # experience received:
  #     1000
  elsif($itemcount{10792} == 1) {
    quest::say("Thank you very much. I have always wanted one of these! Hehehe? Just kidding. I see that you have enchanted this coin. I have placed the final enchantment on it - take it back to Romar.");
    quest::summonitem(10793);
    quest::faction(442,  10);
    quest::faction(291, -10);
    quest::exp(2067182);
  }

} 

File: erudnext/Raine_Beteria.lua
function event_say(e)
	if(e.message:findi("hail")) then 
		if(e.other:GetFaction(e.self) <= 5) then
			e.self:Say("Welcome. I am Raine Beteria.");
		else
			e.self:Say("You are lucky to be standing. Leave here immediately or suffer grave consequences! You are not welcome amongst the Craftkeepers.");
		end
	end
end

function event_trade(e)
	local item_lib = require("items");

	if(e.other:GetFaction(e.self) <= 5) then
		if(item_lib.check_turn_in(e.trade, {item1 = 6339,gold = 50})) then
			e.self:Say("Your silver wand has been fully enchanted. Take it and the pouch of silver dust back to the temple of Ro.");
			e.other:SummonItem(6340);
			e.other:Ding();
			e.other:AddEXP(1000);
		elseif(item_lib.check_turn_in(e.trade, {item1 = 10792})) then
			e.self:Say("Thank you very much. I have always wanted one of these! Hehehe? Just kidding. I see that you have enchanted this coin. I have placed the final enchantment on it - take it back to Romar.");
			e.other:SummonItem(10793);
			e.other:Ding();
			e.other:AddEXP(200000);
		end
	end
	item_lib.return_items(e.self, e.other, e.trade);
end
